### PR TITLE
Bugfix/#2024 Filtering users in chat makes chat only users disappear forever

### LINKED ISSUE #2024: Filtering users in chat makes chat only users disappear forever
**Describe the bug**

**To Reproduce**
- Join #aeolus and look at the chat only group.
- Filter by any name, then remove the filter again

**Expected behavior**
The chat only group should have the same users as before

**Actual behavior**
The chat only group stays empty

**OS**
Linux

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by Marc-Spector**
Fixes #2024 

<s>I wanted to add tests to test the list of users when using filters to avoid these errors in the future, but I have no idea how to do it.</s> 

Added unit tests to avoid these errors in the future

### CODE REVIEW

**Comment by Marc-Spector** [comment]
The PR is ready to review

File: src/main/java/com/faforever/client/chat/UserFilterController.java [hunk 1]
```diff
@@ -177,8 +182,12 @@ boolean isGameStatusMatch(ChatChannelUser chatUser) {
   }
 
   boolean isCountryMatch(ChatChannelUser chatUser) {
+    if (countryFilterField.getText().isEmpty()) {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/UserFilterController.java:185]: Add a comment that this addresses chat-only users for future reference.

File: src/main/java/com/faforever/client/chat/UserFilterController.java [hunk 2]
```diff
@@ -80,13 +80,18 @@ public void filterUsers() {
   }
 
   private boolean filterUser(CategoryOrChatUserListItem userListItem) {
+    if (userListItem.getCategory() != null) {
+
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/chat/UserFilterController.java]: remove empty line

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
Nice catch.

**Review by Brutus5000 — APPROVED** [review]

**Review by Sheikah45 — APPROVED** [review]
looks good

**Pull request merged**